File: filt/lib/Testfile.f90
program Test
  use rsf

  implicit none
  type (file)                      :: in, out
  integer                          :: n1, i
!  character (len=100)              :: label1
  real, dimension (:), allocatable :: trace

  call sf_init()
  in = rsf_input()
  out = rsf_output("out")

  call from_par(in,"n1",n1)
  allocate (trace (n1))

  call to_par(out,"n2",5)

  call rsf_read(in,trace)
  do i=1,5
     call rsf_write(out,trace)
  end do
end program Test

!	$Id: Testfile.f90,v 1.3 2003/09/29 14:34:55 fomels Exp $	

File: filt/lib/Testfile.cc
#include <valarray>

#include "rsf.hh"

int main(int argc, char* argv[])
{
    sf_init(argc,argv);

    iRSF par(0);
    iRSF in;
    oRSF out;
    int n1;
    
    in.get("n1",n1);

    std::valarray<float> trace(n1);

    in >> trace;

    out.put("n2",5);

    for (int i=0; i < 5; i++) {
	out << trace;
    }

    exit (0);
}
